Python/gc_free_threading.c

```diff
@@ -358,7 +358,8 @@ gc_visit_stackref(_PyStackRef stackref)
     // being dead already.
     if (PyStackRef_IsDeferred(stackref) && !PyStackRef_IsNull(stackref)) {
         PyObject *obj = PyStackRef_AsPyObjectBorrow(stackref);
-        if (_PyObject_GC_IS_TRACKED(obj)) {
+        if (_PyObject_GC_IS_TRACKED(obj)
+            && (obj->ob_gc_bits & _PyGC_BITS_FROZEN) == 0) {
```

Comment:
I suggest adding an internal _PyObject_GC_IS_FROZEN() function for the `(obj->ob_gc_bits & _PyGC_BITS_FROZEN)` test.

Reply:
If we refactor this into it's own function, let's do it as a file-local static `gc_is_frozen()` (like `gc_is_unreachable`) instead of a `_PyObject_GC_IS_FROZEN()` in `pycore_gc.h`:

* It's only used in this file -- we don't check frozen status in other `.c` files.
* The other `_PyObject_GC_IS_XXX` functions have implementations for both the default and free threading builds, but that's harder, maybe impractical, for `_PyObject_GC_IS_FROZEN`

Reply:
Ah right, I'm fine with a file-local function.